src/v/cloud_storage/remote.cc

```diff
@@ -106,7 +107,13 @@ static error_outcome categorize_error(
             code.value() != ECONNREFUSED && code.value() != ENETUNREACH
             && code.value() != ETIMEDOUT && code.value() != ECONNRESET
             && code.value() != EPIPE) {
-            vlog(ctxlog.error, "System error {}", cerr);
+            // When a bad file descriptor is seen and the leadership is lost,
+            // downgrade to a warning because gc could have removed the segment
+            if (code.value() == EBADF && !is_leader) {
```

Comment:
TBH the leadership check looks quite out of place at this layer of abstraction - I like the read lock solution better. We can then add the predicate (generically named, something like `should_continue`) as an optimization.